spec/unit/functions/step_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'puppet'
+require 'spec_helper'
+require 'puppet_spec/compiler'
+
+require 'shared_behaviours/iterative_functions'
+
+describe 'the step method' do
+  include PuppetSpec::Compiler
+
+  it 'raises an error when used against an unsupported type' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        3.14.step(1) |$v| {  }
+      MANIFEST
+    end.to raise_error(Puppet::Error, /Cannot be coerced into an Iterable/)
+  end
+
+  it 'raises an error when called with any parameters besides a block' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        [1].step(1,2) |$v| {  }
+      MANIFEST
+    end.to raise_error(Puppet::Error, /expects 2 arguments, got 3/)
+  end
+
+  it 'raises an error when called with something that is not a block' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        [1].step(1,2)
+      MANIFEST
+    end.to raise_error(Puppet::Error, /expects 2 arguments, got 3/)
```

Comment:
as before this test is questionable - unless tested with a call that is constructed in ruby